arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -0,0 +1,1936 @@
+      "type": "object",
+      "properties": {
+        "lockDurationAsIso8601": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "ttlAsIso8601": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "maxDeliveryCount": {
+          "format": "int32",
+          "type": "integer"
+        }
+      }
+    },
+    "CloudToDeviceProperties": {
+      "description": "Contains user facing view of Iot Hub Cloud-To-Device messaging properties",
+      "type": "object",
+      "properties": {
+        "maxDeliveryCount": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "defaultTtlAsIso8601": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "feedback": {
+          "$ref": "#/definitions/FeedbackProperties",
+          "description": "Gets the feedback properties."
+        }
+      }
+    },
+    "OperationsMonitoringProperties": {
+      "description": "Contains the user facing view of the Operation Monitoring properties",
+      "type": "object",
+      "properties": {
+        "events": {
+          "type": "object",
+          "additionalProperties": {
+            "enum": ["None", "Error", "Information"],
+            "type": "string"
+          }
+        }
+      }
+    },
+    "SharedAccessAuthorizationRule": {
+      "type": "object",
+      "properties": {
+        "KeyName": {
+          "type": "string"
+        },
+        "PrimaryKey": {
+          "type": "string"
+        },
+        "IssuerName": {
+          "type": "string"
+        },
+        "SecondaryKey": {
+          "type": "string"
+        },
+        "ClaimType": {
+          "type": "string"
+        },
+        "ClaimValue": {
+          "type": "string"
+        },
+        "Rights": {
+          "type": "array",
+          "items": {
+            "enum": ["Manage", "Send", "Listen", "ManageNotificationHub"],
+            "type": "string"
+          }
+        },
+        "CreatedTime": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "ModifiedTime": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "Revision": {
+          "format": "int64",
+          "type": "integer"
+        }
+      }
+    },
+    "FeedbackProperties": {
+      "description": "Contains user facing view of Iot Hub Feedback Queue properties",
+      "type": "object",
+      "properties": {
+        "lockDurationAsIso8601": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "ttlAsIso8601": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "maxDeliveryCount": {
+          "format": "int32",
+          "type": "integer"
+        }
+      }
+    },
+    "IotHubDescription": {
+      "description": "Contains user facing view of the Description of the IotHub",
+      "type": "object",
+      "properties": {
+        "name": {
```

Comment:
It is advisable to create a Resource model definition and apply the x-ms-azure-resource extension on it. All the resource models can then inherit (allOf) from the Resource model. In that way you do not have to repeat resource properties in every model and reduces the chances of errors. You can carefully make the necessary set of properties as required in the resource object as per your requirement.
Take a look at the storage example: https://github.com/Azure/azure-rest-api-specs/blob/master/arm-storage/2016-01-01/swagger/storage.json#L945

FYI: If a models parent is tagged with "x-ms-azure-resource" extension then it will also be considered as a shared resource.